mlflow/data/delta_dataset_source.py

```diff
@@ -0,0 +1,54 @@
+from typing import TypeVar, Any, Union, Optional, Mapping, Sequence, Dict
+
+from mlflow.data.dataset_source import DatasetSource
+from pyspark.sql import DataFrame
+
+
+DeltaDatasetSourceType = TypeVar("DeltaDatasetSourceType", bound="DeltaDatasetSource")
+
+
+class DeltaDatasetSource(DatasetSource):
+    def __init__(
+        self,
+        path: str,
```

Comment:
We should also support specifying the delta table name and version instead of path